### PR TITLE
No default CS pins for non-SPI drivers

### PR DESCRIPTION
**Pull request by ManuelMcLure**
### Description
On Re-ARM, don't define a default <axis>_CS_PIN unless the driver in question is SPI capable so GPIO pins are not unnecessarily protected.
### Benefits
Allows easier use of the GPIO pins on the Re-ARM J12 connector if SPI capable drivers are not in use.
### Related Issues
None

As part of this fix I moved the ``AXIS_HAS_SPI()`` macro from ``Conditionals_post.h`` to ``drivers.h`` so it's available in the pins files.
I only fixed this for Re-ARM since it's a case where pins on a GPIO connector are auto-declared as CS pins. There are other boards where this applies, but it takes close examination to ensure that the pin in question is available on a GPIO header for this to be useful (for example the Azteeg X5 GT defines CS pins, but they are not available on a GPIO header but only at the driver connections).

### CODE REVIEW

**Comment by ManuelMcLure** [comment]
I went through some of the other pins files and fixed the ones which were obvious. The MKS SBASE pins file needs some work - it already _tries_ to not define CS pins but it only checks for TMC2130 drivers and defines all pins even if only one driver is a 2130. This needs a revamp.

**Review by kAdonis — CHANGES_REQUESTED** [review]
The AXIS_HAS_SPI macro is incomplete, the TMC5130, TMC5160 have also SPI

File: Marlin/src/core/drivers.h [hunk 1]
```diff
@@ -75,7 +75,9 @@
 // Test for supported TMC drivers that require advanced configuration
 // Does not match standalone configurations
 #define HAS_TRINAMIC ( HAS_DRIVER(TMC2130) || HAS_DRIVER(TMC2160) || HAS_DRIVER(TMC2208) || HAS_DRIVER(TMC2660) || HAS_DRIVER(TMC5130) || HAS_DRIVER(TMC5160) )
-
+// Test for a driver that uses SPI - this allows checking whether we need to define a _CS_ pin
+#define AXIS_HAS_SPI(ST)         (AXIS_DRIVER_TYPE(ST, TMC2130) || AXIS_DRIVER_TYPE(ST, TMC2160) || AXIS_DRIVER_TYPE(ST, TMC2660))
```
Comment (kAdonis) [inline Marlin/src/core/drivers.h]:
```suggestion
#define AXIS_HAS_SPI(ST)         (AXIS_DRIVER_TYPE(ST, TMC2130) || AXIS_DRIVER_TYPE(ST, TMC2160) || AXIS_DRIVER_TYPE(ST, TMC2660) || AXIS_DRIVER_TYPE(ST, TMC5130) || AXIS_DRIVER_TYPE(ST, TMC5160))
```

**Comment by thinkyhead** [comment]
It is Marlin's approach to define pins in the pins files even if they are not used. Then when some area of the code would want to use them, it checks to see if the feature is enabled. A sanity check is added to make sure that pins are defined when a feature requires them.

**Comment by thinkyhead** [comment]
The solution to the unnecessary protection problem is to check in `pins.h` (or HAL sensitive pins) whether the pins are in use and therefore need to be included in the `SENSITIVE_PINS` list.

**Comment by ManuelMcLure** [comment]
Fair enough. I'll close this PR and make another that edits ``sensitive_pins.h``. That's a much cleaner solution.

**Pull request closed without merging**